### DISCUSSION TITLE
Pick up environment variables in a custom (event listener) provider

### DISCUSSION THREAD

**Discussion by haiko-ritense** [keycloak/keycloak#21838] [Q&A] [answered]
Hi, 

I am searching for the solution how a custom provider can pick up environment variables from a Docker file. I am trying to use the apporach listed on this page https://www.keycloak.org/server/configuration-provider but it seems not to work. Doe anyone has an example on how to get a variable from a Docker file into a custom provider?

**Comment by danielFesenmeyer** [comment, marked as answer]
Afaik, there are (at least) these two approaches:

- Use the keycloak env var pattern `KC_SPI_<spi-id>_<provider-id>_<property>` and read the value via `config.get("<property>")` in the ProviderFactory's `init(Config.Scope config)` method.
- Use any env vars (not following the Keycloak approach), and just access them with `System.getEnv(name)` in your provider(factory).

**Reply by haiko-ritense** [reply]
Thanks, I have gone for option 2 with System envs
